scipy/optimize/_spectral.py

```diff
@@ -173,6 +191,150 @@ def fmerit(F):
     return result
 
 
+def _fixpoint_squarem(func, x0, args=(), xtol=1e-8, xatol=1e-300,
+                      maxfev=1000, fnorm=None, callback=None, disp=False,
+                      fmerit=None, step_rule=3, **unknown_options):
+    r"""
+    Solve a fixed-point equation with the SQUAREM method.
+
+    Parameters
+    ----------
+    func : callable
+        Function whose fixed point to find
+    x0 : ndarray
+        Initial guess
+    args : tuple
+        Extra parameters to `func`
+    callback : callable, optional
+        Callback function to call on each iteration. Called as
+        ``callable(x, f)`` where ``x`` is the current iterate and
+        ``f`` the residual.
+    xtol : float, optional
+        Relative norm tolerance.
+    xatol : float, optional
+        Absolute norm tolerance.
+        Algorithm terminates when ``||func(x) - x|| < xtol ||x|| + xatol``.
+    fnorm : callable, optional
+        Norm to use in the convergence check. If None, 2-norm is used.
+    maxfev : int, optional
+        Maximum number of function evaluations.
+    disp : bool, optional
+        Whether to print convergence process to stdout.
+    step_rule : {1, 2, 3}, optional
+        Which of the SQUAREM step length rules to use, see below.
+        Default: 2
+    fmerit : callable, optional
+        Merit function to be minimized in the globalization step;
+        called as ``fmerit(x, F)``. For EM, take the negative of
+        the likelihood function. Default: ||F - x||**2
+
+    Notes
+    -----
+    The convergence result for the algorithm in [1] assumes that the function
+    is a contraction in the norm specified by the merit function.
+
+    There are three possible choices for the step length, see [1]_::
+
+        (1)  alpha_k = vdot(r, v) / vdot(v, v)
+        (2)  alpha_k = vdot(r, r) / vdot(r, v)
+        (3)  alpha_k = -sqrt(vdot(r, r) / vdot(v, v))
+
+    The different choices minimize different step length model objective functions.
+
+    References
+    ----------
+    .. [1] R. Varadhan, C. Roland. Scand. J. Statistics, 35, 335 (2008).
```

Comment:
add title ? easier to search for, although google found also by author and journal.